File: leetcode/0001.two_sum/golang/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, n := range nums {
		_, found := m[n]
		if found {
			return []int{m[n], i}
		}
		m[target-n] = i
	}
	return nil
}

func main() {
	indices := twoSum([]int{2, 7, 11, 15}, 9)
	fmt.Println(indices)
}

File: leetcode/0001.two_sum/python/two_sum.py

def twoSum(nums, target):
    if len(nums) <= 1:
        return False
    buff_dict = {}
    for i in range(len(nums)):
        if nums[i] in buff_dict:
            return [buff_dict[nums[i]], i]
        else:
            buff_dict[target - nums[i]] = i

if __name__ == '__main__':
    answer = twoSum([2,7,11,15], 9)
    print(answer)




